libs/sdk-ui-ext/src/internal/components/configurationPanels/WaterfallChartConfigurationPanel.tsx

```diff
@@ -0,0 +1,98 @@
+    SHOW_DELAY_DEFAULT,
+    HIDE_DELAY_DEFAULT,
+    BUBBLE_ARROW_OFFSET_X,
+    BUBBLE_ARROW_OFFSET_Y,
+} from "../../constants/bubble";
+import { messages } from "../../../locales";
+import LegendSection from "../configurationControls/legend/LegendSection";
+
+import BaseChartConfigurationPanel from "./BaseChartConfigurationPanel";
+import { IConfigurationPanelContentProps } from "./ConfigurationPanelContent";
+import TotalSection from "../configurationControls/total/TotalSection";
+
+export interface IWaterfallChartConfigurationPanel extends IConfigurationPanelContentProps {
+    dataLabelDefaultValue?: string | boolean;
+}
+
+export default class WaterfallChartConfigurationPanel extends BaseChartConfigurationPanel<IWaterfallChartConfigurationPanel> {
+    protected renderConfigurationPanel(): React.ReactNode {
+        const { gridEnabled, axes } = this.getControlProperties();
+
+        const { properties, propertiesMeta, pushData, dataLabelDefaultValue = false } = this.props;
+
+        const controlsDisabled = this.isControlDisabled();
+
+        return (
+            <BubbleHoverTrigger showDelay={SHOW_DELAY_DEFAULT} hideDelay={HIDE_DELAY_DEFAULT}>
+                <div>
+                    {this.renderColorSection()}
+                    {this.getBaseChartAxisSection(axes)}
+                    {this.renderLegendSection()}
+
+                    <TotalSection
+                        controlsDisabled={controlsDisabled}
+                        properties={properties}
+                        propertiesMeta={propertiesMeta}
+                        pushData={pushData}
+                    />
+
+                    <ConfigSection
+                        id="canvas_section"
+                        className="gd-canvas-section"
+                        title={messages.canvasTitle.id}
+                        propertiesMeta={propertiesMeta}
+                        properties={properties}
+                        pushData={pushData}
+                    >
+                        <DataLabelsControl
+                            pushData={pushData}
+                            properties={properties}
+                            isDisabled={controlsDisabled}
+                            defaultValue={dataLabelDefaultValue}
+                        />
+
+                        <CheckboxControl
+                            valuePath="grid.enabled"
+                            labelText={messages.canvasGridLine.id}
+                            properties={properties}
+                            checked={gridEnabled}
+                            disabled={controlsDisabled}
+                            pushData={pushData}
+                        />
+                    </ConfigSection>
+                </div>
+                <Bubble
+                    className={this.getBubbleClassNames()}
+                    arrowOffsets={{ "tc bc": [BUBBLE_ARROW_OFFSET_X, BUBBLE_ARROW_OFFSET_Y] }}
+                    alignPoints={[{ align: "tc bc" }]}
```

Comment:
Please, extract objects in `arrowOffsets` and `alignPoints` as constants outside of the class to have them static so reference to object is not changed for each render. The data in object do not rely on any component props or state.

Reply:
I updated this. Thank you!